cpp/test/writer/tsfile_writer_test.cc

```diff
@@ -221,7 +222,80 @@ TEST_F(TsFileWriterTest, WriteMultipleRecords) {
         DataPoint point(measurement_name, (int32_t)i);
         record.append_data_point(point);
         ASSERT_EQ(tsfile_writer_->write_record(record), E_OK);
+        ASSERT_EQ(tsfile_writer_->flush(), E_OK);
+    }
+    ASSERT_EQ(tsfile_writer_->close(), E_OK);
+}
+
+TEST_F(TsFileWriterTest, WriteMultipleTabletsMultiFlush) {
+    const int device_num = 20;
+    const int measurement_num = 20;
+    int max_rows = 100;
+    std::vector<std::vector<MeasurementSchema>> schema_vecs(
+        device_num, std::vector<MeasurementSchema>(measurement_num));
+    for (int i = 0; i < device_num; i++) {
+        std::string device_name = "test_device" + std::to_string(i);
+        for (int j = 0; j < measurement_num; j++) {
+            std::string measure_name = "measurement" + std::to_string(j);
+            schema_vecs[i][j] =
+                MeasurementSchema(measure_name, common::TSDataType::INT32,
+                                  common::TSEncoding::PLAIN,
+                                  common::CompressionType::UNCOMPRESSED);
+            tsfile_writer_->register_timeseries(
+                device_name, measure_name, common::TSDataType::INT32,
+                common::TSEncoding::PLAIN,
+                common::CompressionType::UNCOMPRESSED);
+        }
+    }
+
+    for (int row = 0; row < max_rows; row++) {
+        for (int i = 0; i < device_num; i++) {
+            std::string device_name = "test_device" + std::to_string(i);
+            Tablet tablet(device_name, &schema_vecs[i], max_rows);
+            tablet.init();
+            for (int j = 0; j < measurement_num; j++) {
+                tablet.set_timestamp(row, 16225600000 + row * 100);
+                tablet.set_value(row, j, static_cast<int32_t>(row));
+            }
+            ASSERT_EQ(tsfile_writer_->write_tablet(tablet), E_OK);
+        }
+        ASSERT_EQ(tsfile_writer_->flush(), E_OK);
+    }
```

Comment:
max_rows should be the inner loop. Otherwise, each tablet will only have one row but will be declared with max_rows rows.